### PR TITLE
Version 2

### PR DESCRIPTION
**Pull request by Massi-X**
Hi max :P

### CODE REVIEW

**Comment by Massi-X** [comment]
OK this is all!
PS. you can ignore also #6034846

**Comment by Massi-X** [comment]
I'll never stop :P
I made this last commit whit the possibility to choose from your implementation and mine for the track selector.
And I've also improved my version.

File: app/src/main/res/xml/preferences.xml [hunk 1]
```diff
@@ -78,4 +92,4 @@
     <Preference
         android:key="donate"
         android:title="@string/donate" />
-</PreferenceScreen>
\ No newline at end of file
+</PreferenceScreen>
```
Comment (Maxr1998) [inline app/src/main/res/xml/preferences.xml:95]: (Note to self: remove empty line)

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -62,4 +62,12 @@
...
 
     <string name="donate">Donate to the developer</string>
+    <string name="np_voice_control">Add Voice Commands</string>
+    <string name="np_voice_control_summary">Add voice commands shortcut next to the queue button\nNote: Overwritten by equalizer shortcut</string>
+    <string name="np_no_album_art">No Album Arts in Track Selector</string>
+    <string name="np_no_album_art_summary">Loading album arts require some time, so you can disable it here.\nThe list is also limited to 40 songs.</string>
+    <string name="xposed_description">Tweak the layout of Google Play Music</string>
+    <string name="open_GPM">Open GPM</string>
+    <string name="np_alternative_art_loader">Alternative Art Loader</string>
+    <string name="np_alternative_art_loader_summary">If normally you can\'t see album arts in the track selector, you can try this.</string>
```
Comment (Maxr1998) [inline app/src/main/res/values/strings.xml:72]: (Note to self: put into right order)

File: app/src/main/res/values-it/strings.xml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    <string name="np_voice_control">Aggiungi comandi vocali</string>
+    <string name="np_voice_control_summary">Aggiungi i comandi vocali nella schermata di riproduzione\nNB. Sovrascritto dal link all\'equalizzatore</string>
+    <string name="np_no_album_art">Nessuna cover nel selettore di tracce</string>
+    <string name="np_no_album_art_summary">Il caricamento delle cover richiede qualche tempo, perciò puoi disabilitarlo.\nInoltre la lista è limitata a 40 canzoni.</string>
+    <string name="xposed_description">Varie modifiche per Google Play Music</string>
+    <string name="open_GPM">Apri GPM</string>
+    <string name="np_alternative_art_loader">Caricamento Cover Alternativo</string>
+    <string name="np_alternative_art_loader_summary">Se normalmente non vedi le cover nel selettore di tracce, puoi provare questo.</string>
+</resources>
```
Comment (Maxr1998) [inline app/src/main/res/values-it/strings.xml:74]: Awesome, thanks!

File: app/src/main/java/de/Maxr1998/xposed/gpm/hooks/NowPlaying.java [hunk 4]
```diff
@@ -115,7 +115,9 @@ protected void afterHookedMethod(MethodHookParam param) throws Throwable {
                 protected void afterHookedMethod(final MethodHookParam param) throws Throwable {
                     PREFS.reload();
                     View queueSwitcher = (View) getObjectField(param.thisObject, "mQueueSwitcher");
-                    ((View) queueSwitcher.getParent()).findViewById(EQ_BUTTON_ID).setVisibility(queueSwitcher.getVisibility());
+                    View EQ_BUTTON_ID_TMP = ((View) queueSwitcher.getParent()).findViewById(EQ_BUTTON_ID);
```
Comment (Maxr1998) [inline app/src/main/java/de/Maxr1998/xposed/gpm/hooks/NowPlaying.java:118]: (Note to self: rename view to no caps)

File: app/src/main/java/de/Maxr1998/xposed/gpm/hooks/NotificationMod.java [hunk 5]
```diff
@@ -219,4 +349,4 @@ protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
             log(t);
         }
     }
-}
\ No newline at end of file
+}
```
Comment (Maxr1998) [inline app/src/main/java/de/Maxr1998/xposed/gpm/hooks/NotificationMod.java:352]: (Note to self: remove empty line)

File: app/src/main/java/de/Maxr1998/xposed/gpm/hooks/Main.java [hunk 6]
```diff
@@ -93,4 +93,4 @@ public void handleInitPackageResources(XC_InitPackageResources.InitPackageResour
             NowPlaying.initResources(resParam);
         }
     }
-}
\ No newline at end of file
+}
```
Comment (Maxr1998) [inline app/src/main/java/de/Maxr1998/xposed/gpm/hooks/Main.java:96]: (Note to self: remove empty line)

File: app/src/main/java/de/Maxr1998/xposed/gpm/hooks/Main.java [hunk 7]
```diff
@@ -67,7 +67,7 @@ private void doHooks(XC_LoadPackage.LoadPackageParam lPParam) {
         findAndHookMethod(GPM + ".Feature", lPParam.classLoader, "isSnappleEnabled", Context.class, new XC_MethodReplacement() {
             @Override
             protected Object replaceHookedMethod(MethodHookParam param) throws Throwable {
-                return true;
+                return PREFS.getBoolean(Common.NP_VOICE_CONTROL, false);
```
Comment (Maxr1998) [inline app/src/main/java/de/Maxr1998/xposed/gpm/hooks/Main.java]: Good idea!

File: app/build.gradle [hunk 8]
```diff
@@ -59,4 +59,4 @@ dependencies {
     compile 'de.Maxr1998:track-selector-lib:1.1'
     provided 'de.robv.android.xposed:api:81'
     provided 'de.robv.android.xposed:api:81:sources'
-}
\ No newline at end of file
+}
```
Comment (Maxr1998) [inline app/build.gradle:62]: (Note to self: remove empty line)

**Comment by Maxr1998** [comment]
I have some questions regarding your code, can I contact you via some instant messenger?

**Comment by Massi-X** [comment]
If yuo want we can talk here! I leave the page open and i will check!

**Comment by Massi-X** [comment]
I odont have nothing apart from whatsapp

**Comment by Maxr1998** [comment]
WhatsApp is fine for me, I will send you my number on XDA.

**Comment by Massi-X** [comment]
Ok!

**Comment by Maxr1998** [comment]
@Massi-X Done!

**Pull request merged**